### PR TITLE
Tune List.init threshold

### PR DESCRIPTION
**Pull request by hhugo**
fix https://caml.inria.fr/mantis/view.php?id=7764

### CODE REVIEW

**Comment by Gbury** [comment]
Would it make sense for this kind of adaptive constant to also apply to the new `List.of_seq` function (maybe with different values) ?
cc @c-cube

**Review by gasche — APPROVED** [review]
Could you move the `.depend` changes to a different commit? This would help rebasing / cherry-picking / backporting etc.

**Comment by hhugo** [comment]
@gasche, done

File: stdlib/list.ml [hunk 1]
```diff
@@ -66,9 +66,15 @@ let rec init_aux i n f =
     let r = f i in
     r :: init_aux (i+1) n f
 
+let rev_init_threshold =
+  match Sys.backend_type with
+  | Sys.Native | Sys.Bytecode -> 10_0000
```
Comment (redianthus) [inline stdlib/list.ml:71]: @gasche, @hhugo, I guess it was supposed to be `10_000`, not `10_0000`...

Comment (gasche) [reply stdlib/list.ml:71]: Indeed, good catch... It is a bit disconcerting that our CI didn't catch that regression.

Comment (gasche) [reply stdlib/list.ml:71]: I fixed this in fa884f420b257f350a5a4a3e701b5acd1df9c10a.

**Pull request merged**